modules/alloy/src/test/ts/browser/ui/typeahead/TypeaheadWithDifferentMothershipsTest.ts

```diff
@@ -0,0 +1,215 @@
+
+import { ApproxStructure, Assertions, Keyboard, Keys, Mouse, UiControls, UiFinder } from '@ephox/agar';
+import { after, before, context, describe, it } from '@ephox/bedrock-client';
+import { Objects } from '@ephox/boulder';
+import { Arr, Future, Optional, Result } from '@ephox/katamari';
+import { SugarDocument, Value } from '@ephox/sugar';
+
+import { Focusing } from 'ephox/alloy/api/behaviour/Focusing';
+import { AlloyComponent } from 'ephox/alloy/api/component/ComponentApi';
+import * as GuiFactory from 'ephox/alloy/api/component/GuiFactory';
+import { SketchSpec } from 'ephox/alloy/api/component/SpecTypes';
+import * as AlloyTriggers from 'ephox/alloy/api/events/AlloyTriggers';
+import * as NativeEvents from 'ephox/alloy/api/events/NativeEvents';
+import * as Attachment from 'ephox/alloy/api/system/Attachment';
+import * as Gui from 'ephox/alloy/api/system/Gui';
+import * as GuiSetup from 'ephox/alloy/api/testhelpers/GuiSetup';
+import { TestStore } from 'ephox/alloy/api/testhelpers/TestStore';
+import { tieredMenu as TieredMenu } from 'ephox/alloy/api/ui/TieredMenu';
+import { Typeahead } from 'ephox/alloy/api/ui/Typeahead';
+import * as MenuEvents from 'ephox/alloy/menu/util/MenuEvents';
+import * as TestDropdownMenu from 'ephox/alloy/test/dropdown/TestDropdownMenu';
+import { TestItem } from 'ephox/alloy/test/dropdown/TestDropdownMenu';
+import * as Sinks from 'ephox/alloy/test/Sinks';
+
+// This test checks that all of Typeahead's functions work when the sink is in a
+// different mothership.
+const makeTypeahead = (store: TestStore, sinkFromOtherMothership: AlloyComponent): SketchSpec => {
+  return Typeahead.sketch({
+    inputClasses: [ 'test-typeahead' ],
+    components: [ ],
+
+    minChars: 2,
+    model: {
+      // Enabling populateFromBrowse and selectsOver makes it easier to check it is
+      // all wired together properly.
+      selectsOver: false,
+      populateFromBrowse: true
+    },
+    fetch: (input) => {
+      const text = Value.get(input.element).toLowerCase();
+
+      const items = Arr.map(
+        Array(3),
+        (x, i): TestItem => {
+          return {
+            type: 'item',
+            data: {
+              value: `item-${i}-value`,
+              meta: {
+                text: `Item-${i}-text`
+              }
+            }
+          };
+        }
+      );
+
+      const future = Future.pure<TestItem[]>(items);
+
+      return future.map((f) => {
+        // TODO: Test this.
```

Comment:
Test what more exactly?

Reply:
I think it was a comment that came along for the ride. I'd say it was the "no-data" path.

Reply:
- [ ] remove

Reply:
Updated: removed comment.